File: java/src/quick/quick.java
package quick;

import java.util.Arrays;

public class quick{
    public static void quickSort(int []nums, int left, int right){
        if (left >= right){
            return;
        }
        int i = left - 1;
        int j = right + 1;
        int x = nums[left];
        while (i < j) {
            while (nums[++i] < x);
            while (nums[--j] > x);
            if (i < j) {
                int tmp = nums[i];
                nums[i] = nums[j];
                nums[j] = tmp;
            }
        }
        quickSort(nums, left, j);
        quickSort(nums, j+1, right);
    }
    public static void main(String[] args) {
        int []nums = {2, 4, 3, 6, 5};
        quickSort(nums, 0, nums.length-1);
        System.out.println(Arrays.toString(nums));
    }
}

File: go/src/quick/quick.go
package main

import "fmt"

func quickSort(nums []int, left, high int){
	if left >= high {
		return 
	}
	i, j := left-1, high+1
	x := nums[(left+high)>>1]
	for i < j {
		for {
			i++
			if nums[i] >= x {
				break
			}
		}
		for {
			j--
			if nums[j] <= x{
				break
			}
		}
		if i < j {
			nums[i], nums[j] = nums[j], nums[i]
		}
	}
	quickSort(nums, left, j)
	quickSort(nums, j+1, high)
}

func main(){
	nums := []int{2, 5, 6, 4, 1, 7}
	quickSort(nums, 0, len(nums)-1)
	fmt.Println(nums)
}